File: drills/ts/harnesses/nc100_edit_distance.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.minDistance !== "function") throw new Error("__signature_mismatch__");
  return [
    ["minDistance('horse', 'ros')", show(3), show(solution.minDistance('horse', 'ros'))],
    ["minDistance('intention', 'execution')", show(5), show(solution.minDistance('intention', 'execution'))],
    ["minDistance('', 'abc')", show(3), show(solution.minDistance('', 'abc'))],
    ["minDistance('abc', '')", show(3), show(solution.minDistance('abc', ''))],
    ["minDistance('abc', 'abc')", show(0), show(solution.minDistance('abc', 'abc'))],
    ["minDistance('kitten', 'sitting')", show(3), show(solution.minDistance('kitten', 'sitting'))],
  ];
}

File: drills/elixir/harnesses/nc100_edit_distance.exs
[
  {~S{min_distance("horse", "ros")}, inspect(3), inspect(Solution.min_distance("horse", "ros"))},
  {~S{min_distance("intention", "execution")}, inspect(5), inspect(Solution.min_distance("intention", "execution"))},
  {~S{min_distance("", "abc")}, inspect(3), inspect(Solution.min_distance("", "abc"))},
  {~S{min_distance("abc", "")}, inspect(3), inspect(Solution.min_distance("abc", ""))},
  {~S{min_distance("abc", "abc")}, inspect(0), inspect(Solution.min_distance("abc", "abc"))},
  {~S{min_distance("kitten", "sitting")}, inspect(3), inspect(Solution.min_distance("kitten", "sitting"))}
]
